src/batch_group/entity.rs

```diff
@@ -55,13 +55,17 @@ impl BatchGroup {
     }
 
     pub fn update_description(&mut self, description: String) {
-        self.events
-            .push(BatchGroupEvent::BatchGroupDescriptionUpdated { description });
+        if self.description() != Some(description.clone()) {
```

Comment:
Let's try to avoid cloning as much as possible. `if self.description().as_ref() != Some(&description)` should work